src/Network/Receive/iRO.pm

```diff
@@ -42,6 +45,92 @@ sub new {
 	return $self;
 }
 
+=pod
+struct packet_achievement_update {
+	uint16 packet_id;  2
+	uint32 total_points; 4
+	uint16 rank; 2
+	uint32 current_rank_points; 4
+	uint32 next_rank_points; 4
+	struct ach_list_info ach;
+} __attribute__((packed));
+
+struct ach_list_info {
+	uint32 ach_id; 4
+	uint8 completed; 1
+	uint32 objective[MAX_ACHIEVEMENT_OBJECTIVES]; 4*10
+	uint32 completed_at; 4
+	uint8 reward; 1
+} __attribute__((packed));
+
+struct packet_achievement_list {
+	uint16 packet_id;
+	uint16 packet_len;
+	uint32 total_achievements;
+	uint32 total_points;
+	uint16 rank;
+	uint32 current_rank_points;
+	uint32 next_rank_points;
+	struct ach_list_info ach[MAX_ACHIEVEMENT_DB];
+} __attribute__((packed));
+
+struct packet_achievement_reward_ack {
+	uint16 packet_id;
+	uint8 received;
+	uint32 ach_id;
+} __attribute__((packed));
+=cut
+
+sub achievement_list {
+	my ($self, $args) = @_;
+	
+	my $msg = $args->{RAW_MSG};
+	my $msg_size = $args->{RAW_MSG_SIZE};
+	my $headerlen = 22;
+	my $achive_len = 50;
```

Comment:
I like doing something like:

```
my $achieve_pack = 'V C V10 V C';
my $achieve_len = pack $achieve_pack;
```

... for less magic variables. What do you think?